src/helpers/passport.ts

```diff
@@ -1,55 +1,51 @@
-import * as passport from 'passport'
-import * as jwt from 'jsonwebtoken'
-import conf from '../config'
-import { Strategy, ExtractJwt } from 'passport-jwt'
-import { User } from '../sequelize'
+import * as passport from 'passport';
+import * as jwt from 'jsonwebtoken';
+import conf from '../config';
+import { Strategy, ExtractJwt } from 'passport-jwt';
+import { User } from '../sequelize';
 
-export function config(app: any) {
-  app.use(passport.initialize())
+export function config(app: any): void {
+  app.use(passport.initialize());
 
-  passport.use(new Strategy({
+  passport.use(new Strategy(
+    {
       secretOrKey: conf.jwt.secret,
       jwtFromRequest: ExtractJwt.fromAuthHeader(),
     },
     (payload, next) => {
-      User.findOne({where: {'id': payload.id}})
+      User.findOne({ where: { id: payload.id } })
       .then(user => {
-        if (user) {
-          next(null, user)
-        } else {
-          next(null, false)
-        }
+        if (user) return next(null, user);
+        next(null, false);
       })
-      .catch(err => next(null, false, {message: err}))
+      .catch(err => next(null, false, { message: err }));
     }
-  ))
+  ));
 }
 
-export function authorize(req, res, next) {
+export function authorize(req: any, res: any, next: any): Promise<any> {
   return new Promise((resolve, reject) => {
-    passport.authenticate('jwt', {session: conf.jwt.session},
-      function(err, user, info) {
-        if (err || !user) {
-          reject(info)
-        }
-        resolve(user)
+    passport.authenticate(
+      'jwt',
+      { session: conf.jwt.session },
+      (err, user, info) => {
+        if (err || !user) return reject(info);
+        resolve(user);
```

Comment:
Here `passport` uses non-promise API, so I have to create and return my own `Promise` as it's described in an above article `Rookie mistake #4: using "deferred"`. In comment below I should omit my own `Promise` and return chain from `sequelize` request.